src/http/controllers/products-controllers.ts

```diff
@@ -29,9 +42,19 @@ export const updateProduct = async (ctx, next) => {
     const product: Product = await queries.updateProductQuery(
         ctx.params.id,
         ctx.request.body
-    );
+    ) as Product;
     ctx.ok(product);
     await next();
 };
 
-
+export const checkout = async (ctx, next) => {
+    const usersProducts = UserCart.getUsersProducts();
+    const userId = ctx.request.body.id;
+    await Promise.all(
+        Object.keys(usersProducts[userId]).map(async productId => {
+                await queries.checkoutProductQuery(productId, usersProducts[userId][productId], ctx);
+                UserCart.deleteUserCart(userId);
```

Comment:
why do you delete the cart so many time?

Reply:
i dont want the information to be stored in the cart after checkout beacuse it is already in db. if it will stay in cart there will be duplicate calculation of products left